syntax = "proto3";
package protoData;
message GoodsArray {
    repeated Goods goods = 1;
}
message Goods{
    string name = 1;
    string description = 2;
    double price = 3;
    int32 seller = 4;// primary key of seller
    int32 id = 5;
    int32 type = 6;
    int32 remain = 7;
}

message User{
    int32 id = 1;
    string username = 2;
    string password = 3;
    double balance = 4;
    int32 type = 5;
}
message UserForm{
    string username = 1;
    string password = 2;
    int32 type = 3;
}
message TransactionArray{
    repeated Transaction transaction = 1;
}

message Transaction{
    int32 id = 1;
    int32 from = 2;
    int32 to = 3;
    string from_name = 4;
    string to_name = 5;
    bool finished = 6;
    double volume = 7;
    string detail = 8;
    int64 timestamp = 9;
}
message CartItem{
    int32 user = 1;
    int32 goods = 2;
    int32 count = 3; 
}
message Cart{
    repeated CartItem item = 1;
}

message Discount{
    int32 id = 1;
    int32 type = 2;
    int32 operand = 3;
    double disocunt = 4;
    double threshold = 5;
}
